pkg/kubelet/status/manager.go

```diff
@@ -306,7 +305,12 @@ func (m *manager) RemoveOrphanedStatuses(podUIDs map[types.UID]bool) {
 }
 
 // syncBatch syncs pods statuses with the apiserver.
-func (m *manager) syncBatch() {
+// The parameter isReconcileEnabled is here mostly for unit test. After adding reconciliation,
+// if status in pod manager is different from the cached status, a reconciliation will be triggered,
+// which will mess up all the old unit test.
```

Comment:
Rather than complicating the production code with an extra parameter, you could create a fake status manager in the test and override needsReconcile to always return false.